core/domain/rights_domain.py

```diff
@@ -231,6 +235,49 @@ def is_solely_owned_by_user(self, user_id):
         """
         return user_id in self.owner_ids and len(self.owner_ids) == 1
 
+    def assign_role(self, user_id, new_role):
```

Comment:
The name doesn't indicate that anything is returned. Rename it to something that implies that the old role is returned.

Reply:
Done